requests/sessions.py

```
@@ -466,28 +354,135 @@ def delete(self, url, **kwargs):
 
         return self.request('DELETE', url, **kwargs)
 
-    def send(self, request, **kwargs):
-        """Send a given PreparedRequest."""
-        # Set defaults that the hooks can utilize to ensure they always have
-        # the correct parameters to reproduce the previous request.
-        kwargs.setdefault('stream', self.stream)
-        kwargs.setdefault('verify', self.verify)
-        kwargs.setdefault('cert', self.cert)
-        kwargs.setdefault('proxies', self.proxies)
+    def resolve_one_redirect(self, resp):
+        """Given a response known to be a redirect, determine the
+           destination of the redirect, fire a new request for that
+           URL, and produce a new response object.
 
-        # It's possible that users might accidentally send a Request object.
-        # Guard against that specific failure case.
-        if not isinstance(request, PreparedRequest):
-            raise ValueError('You can only send PreparedRequests.')
+           :param resp: The redirection to be resolved.
+        """
 
-        # Set up variables needed for resolve_redirects and dispatching of hooks
-        allow_redirects = kwargs.pop('allow_redirects', True)
-        stream = kwargs.get('stream')
-        timeout = kwargs.get('timeout')
-        verify = kwargs.get('verify')
-        cert = kwargs.get('cert')
-        proxies = kwargs.get('proxies')
-        hooks = request.hooks
+        # As a safety feature, if passed a non-redirect, do nothing.
+        if not resp.is_redirect:
+            return resp
```

I would be inclined to throw an exception here, rather than silently refuse to do anything. `ValueError` maybe?